src/cephadm/cephadm

```diff
@@ -1853,6 +1853,9 @@ def deploy_daemon_units(fsid, uid, gid, daemon_type, daemon_id, c,
             install_path = find_program('install')
             f.write('{install_path} -d -m0770 -o {uid} -g {gid} /var/run/ceph/{fsid}\n'.format(install_path=install_path, fsid=fsid, uid=uid, gid=gid))
 
+        # Sometimes, adding `--rm` to a run_cmd doesn't work. Let's remove the container manually
```

Comment:
Do we know why this sometimes doesn't work? It doesn't feel like we've identified the root cause here ..

Reply:
which is correct, yes. Looks like podman doesn't honor `--rm`, if the container creation itself failed. This is a workaround for the broken `--rm` handling. 

However, this doesn't solve, why w're getting a ceph-mon: no such file or directory.

Reply:
@mgfritch @sebastian-philipp we also suffered from some issues in older releases of podman and fuse-overlay but they were addressed in newer releases. see https://tracker.ceph.com/issues/44242 , https://github.com/ceph/ceph-container/commit/c44202c2eadef7262468508639644ee32ded0e07 and https://github.com/ceph/ceph-container/commit/6285bae77c3a527f872328d3e2c556eed7879475 . i think our CI (ceph-build and ceph-container) has got them addressed by installing the latest build of podman and its dependencies.

i am wondering if we should do the same in our testing harness as well?

Reply:
hm. you're sure this is related?

Reply:
related, sage closed https://github.com/ceph/ceph/pull/33367 a while ago without merging it.

Reply:
i am not sure. i just want to rule out whatever issues related to podman and its dependencies. if i am sure, i will put the analysis here. but i am not familiar with podman or docker at this moment, so this is what i can do.

Reply:
my idea is to just merge this one and see if it helps.